pkg/cassandra/encryption.go

```diff
@@ -0,0 +1,217 @@
+	}
+
+	// Find the cassandra container and add the volume mounts for both stores
+	addEncryptionMountToCassandra(template, volumes["keystore"], volumes["truststore"], storeType)
+}
+
+func EncryptionVolumes(storeType string, encryptionStores encryption.EncryptionStores) map[string]corev1.Volume {
+	volumes := map[string]corev1.Volume{}
+
+	// Create the volume for the keystore
+	keystoreVolume := corev1.Volume{
+		Name: fmt.Sprintf("%s-keystore", storeType),
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: encryptionStores.KeystoreSecretRef.Name,
+				Items: []corev1.KeyToPath{
+					{
+						Key:  "keystore",
+						Path: "keystore",
+					},
+				},
+			},
+		},
+	}
+
+	// Create the volume for the truststore
+	truststoreVolume := corev1.Volume{
+		Name: fmt.Sprintf("%s-truststore", storeType),
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: encryptionStores.TruststoreSecretRef.Name,
+				Items: []corev1.KeyToPath{
+					{
+						Key:  "truststore",
+						Path: "truststore",
+					},
+				},
+			},
+		},
+	}
+
+	volumes["keystore"] = keystoreVolume
+	volumes["truststore"] = truststoreVolume
+	return volumes
+}
+
+// Adds keystore and truststore volume mounts to the cassandra container.
+func addEncryptionMountToCassandra(template *DatacenterConfig, keystoreVolume, truststoreVolume corev1.Volume, storeType string) {
+	// Initialize the container array if it doesn't exist yet
+	if template.PodTemplateSpec.Spec.Containers == nil {
+		template.PodTemplateSpec.Spec.Containers = make([]corev1.Container, 0)
+	}
+
+	cassandraContainer := &corev1.Container{
+		Name: "cassandra",
+	}
+
+	cassandraContainerIdx, foundCassandra := FindContainer(template.PodTemplateSpec, "cassandra")
+	if foundCassandra {
+		cassandraContainer = template.PodTemplateSpec.Spec.Containers[cassandraContainerIdx].DeepCopy()
+	}
+
+	AddOrUpdateVolumeMount(cassandraContainer, &keystoreVolume, StoreMountFullPath(storeType, "keystore"))
+	AddOrUpdateVolumeMount(cassandraContainer, &truststoreVolume, StoreMountFullPath(storeType, "truststore"))
+
+	// If the container is not found, add it to the pod spec
+	// If the container is found, update it
+	if foundCassandra {
+		template.PodTemplateSpec.Spec.Containers[cassandraContainerIdx] = *cassandraContainer
+	} else {
+		template.PodTemplateSpec.Spec.Containers = append(template.PodTemplateSpec.Spec.Containers, *cassandraContainer)
+	}
+}
+
+func checkMandatoryEncryptionFields(encryptionStores *encryption.EncryptionStores) error {
+	if encryptionStores == nil {
+		return fmt.Errorf("EncryptionStores is required to set up encryption")
+	}
+	if encryptionStores.KeystoreSecretRef.Name == "" {
+		return fmt.Errorf("keystore secret ref was not set")
+	}
+	if encryptionStores.KeystorePasswordSecretRef.Name == "" {
+		return fmt.Errorf("keystore password secret ref was not set")
+	}
+	if encryptionStores.TruststoreSecretRef.Name == "" {
```

Comment:
I'm not sure here, I think the truststore is only required if `require_client_auth` is true.

Reply:
As we've seen, it's not the case :(